metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py

```diff
@@ -162,84 +162,18 @@ def create_checkpoint(self, job_id: JobId) -> Optional[Checkpoint]:
         """
         Create the custom checkpoint with empty state for the job.
         """
-        assert self.ctx.pipeline_name
         if job_id == self.get_default_ingestion_job_id():
-            return Checkpoint(
-                job_name=job_id,
-                pipeline_name=self.ctx.pipeline_name,
-                platform_instance_id=self.get_platform_instance_id(),
-                run_id=self.ctx.run_id,
-                config=self.config,
-                state=BaseUsageCheckpointState(
-                    begin_timestamp_millis=int(
-                        self.config.start_time.timestamp() * 1000
-                    ),
-                    end_timestamp_millis=int(self.config.end_time.timestamp() * 1000),
-                ),
+            return self.redundant_run_skip_handler.create_checkpoint(
+                start_time_millis=datetime_to_ts_millis(self.config.start_time),
+                end_time_millis=datetime_to_ts_millis(self.config.end_time),
             )
         return None
 
-    def _should_skip_this_run(self) -> bool:
-        # Check if forced rerun.
-        if (
-            self.config.stateful_ingestion
-            and self.config.stateful_ingestion.ignore_old_state
-        ):
-            return False
-        # Determine from the last check point state
-        last_successful_pipeline_run_end_time_millis: Optional[int] = None
-        last_checkpoint = self.get_last_checkpoint(
-            self.get_default_ingestion_job_id(), BaseUsageCheckpointState
-        )
-        if last_checkpoint and last_checkpoint.state:
-            state = cast(BaseUsageCheckpointState, last_checkpoint.state)
-            last_successful_pipeline_run_end_time_millis = state.end_timestamp_millis
-
-        if (
-            last_successful_pipeline_run_end_time_millis is not None
-            and int(self.config.start_time.timestamp() * 1000)
-            <= last_successful_pipeline_run_end_time_millis
-        ):
-            warn_msg = (
-                f"Skippig this run, since the last run's bucket duration end: "
-                f"{datetime.fromtimestamp(last_successful_pipeline_run_end_time_millis/1000, tz=timezone.utc)}"
-                f" is later than the current start_time: {self.config.start_time}"
-            )
-            logger.warning(warn_msg)
-            self.report.report_warning("skip-run", warn_msg)
-            return True
+    def is_checkpointing_enabled(self, job_id: JobId) -> bool:
```

Comment:
would it be possible to get rid of this method from the source classes as well?

Reply:
Done!